deployments/sandbox/environment.py

```diff
@@ -55,17 +55,17 @@ def env() -> Mapping[str, Optional[str]]:
         # FIXME: Add tooling to aid in prefix choice
         #        https://github.com/DataBiosphere/azul/issues/3027
         'AZUL_TDR_SOURCES': ','.join([
-            'tdr:broad-jade-dev-data:snapshot/hca_dev_20201203___20210426:42',
+            'tdr:broad-jade-dev-data:snapshot/hca_dev_20201203___20210426:42:2',
         ]),
         **{
             f'AZUL_TDR_{catalog.upper()}_SOURCES': ','.join([
-                f'tdr:broad-jade-dev-data:snapshot/hca_dev_20201023_ebiv4___20210302:4'
+                f'tdr:broad-jade-dev-data:snapshot/hca_dev_20201023_ebiv4___20210302:4:2'
             ])
             for catalog in ('dcp2ebi', 'it2ebi')
         },
         **{
             f'AZUL_TDR_{catalog.upper()}_SOURCES': ','.join([
-                'tdr:broad-jade-dev-data:snapshot/lungmap_dev_20210412__20210414:',
+                'tdr:broad-jade-dev-data:snapshot/lungmap_dev_20210412__20210414::2',
```

Comment:
Same here. Is a partition prefix length of 0 supported? If it is easy to do I would want to include support for it and consider lungmap as a test case of that support.

Reply:
partition_prefix_length of `0` would send one notification to the indexer lambda to remote reindex the entire prefix by itself.
```
> list(map(''.join, itertools.product('0123456789abcdef', repeat=0)))
['']
```

A notification would be passed to `remote_reindex_partition` where the bundles under the prefix + the empty partition would be listed out, then queued up for indexing.

This might defeat the purpose of a remote reindex as we want the bundle enumeration to be parallelized by the lambdas. Rather than preventing this, we can add a warning if the partition_prefix_length is set to `0`. 

> consider lungmap as a test case of that support.

For testing would you like to see another IT, or just confirm that the expected notifications will be passed to the lambdas in a unit test? I think a unit test will be sufficient as the rest of the code has coverage.

What about when the lungmap source changes?

Reply:
> partition_prefix_length of `0` would send one notification to the indexer lambda to remote reindex the entire prefix by itself.

That's the desired behavior.

> A notification would be passed to `remote_reindex_partition` where the bundles under the prefix + the empty partition would be listed out, then queued up for indexing.
> 
> This might defeat the purpose of a remote reindex as we want the bundle enumeration to be parallelized by the lambdas.

I don't think it does. If the source is so small that it can be handled by one lambda invocation, there is no need to further break it up. Partitions smaller than a certain size aren't going to deliver a speed-up. I don't know what that size is, but a partition size that packs one lambda invocation is a good approximation. And even if there is just one partition, remote reindexing makes sense, because it still gets to benefit from the implicit retry mechanism and the fact that the diagnostic output is kept in a central place.

> What about when the lungmap source changes?

That's no different to the question about what happens when any source changes. The person configuring the source needs to know how many bundles it contains in order to make a decision on which partition prefix length to configure. The source is immutable, so a valid choice will always remain valid.